src/traces/scattergl/convert.js

```diff
@@ -0,0 +1,397 @@
+            optsOut.markers[i] = getSymbolSdf(optsIn.symbol[i]);
+        }
+    } else {
+        optsOut.marker = getSymbolSdf(optsIn.symbol);
+    }
+
+    // prepare sizes
+    var markerSizeFunc = makeBubbleSizeFn(trace);
+    var s;
+
+    if(multiSize || multiLineWidth) {
+        var sizes = optsOut.sizes = new Array(count);
+        var borderSizes = optsOut.borderSizes = new Array(count);
+        var sizeTotal = 0;
+        var sizeAvg;
+
+        if(multiSize) {
+            for(i = 0; i < count; i++) {
+                sizes[i] = markerSizeFunc(optsIn.size[i]);
+                sizeTotal += sizes[i];
+            }
+            sizeAvg = sizeTotal / count;
+        } else {
+            s = markerSizeFunc(optsIn.size);
+            for(i = 0; i < count; i++) {
+                sizes[i] = s;
+            }
+        }
+
+        // See  https://github.com/plotly/plotly.js/pull/1781#discussion_r121820798
+        if(multiLineWidth) {
+            for(i = 0; i < count; i++) {
+                borderSizes[i] = markerSizeFunc(optsIn.line.width[i]);
+            }
+        } else {
+            s = markerSizeFunc(optsIn.line.width);
+            for(i = 0; i < count; i++) {
+                borderSizes[i] = s;
+            }
+        }
+
+        optsOut.sizeAvg = sizeAvg;
+    } else {
+        optsOut.size = markerSizeFunc(optsIn && optsIn.size || 10);
+        optsOut.borderSizes = markerSizeFunc(optsIn.line.width);
+    }
+
+    return optsOut;
+}
+
+function convertMarkerSelection(trace, target) {
+    var optsIn = trace.marker;
+    var optsOut = {};
+
+    if(!target) return optsOut;
+
+    if(target.marker && target.marker.symbol) {
+        optsOut = convertMarkerStyle(Lib.extendFlat({}, optsIn, target.marker));
+    } else if(target.marker) {
+        if(target.marker.size) optsOut.sizes = target.marker.size;
+        if(target.marker.color) optsOut.colors = target.marker.color;
+        if(target.marker.opacity !== undefined) optsOut.opacity = target.marker.opacity;
+    }
+
+    return optsOut;
+}
+
+function convertErrorBarStyle(trace, target) {
+    var optsOut = {
+        capSize: target.width * 2,
+        lineWidth: target.thickness,
+        color: target.color
+    };
+
+    if(target.copy_ystyle) {
+        optsOut = trace.error_y;
+    }
+
+    return optsOut;
+}
+
+var SYMBOL_SDF_SIZE = constants.SYMBOL_SDF_SIZE;
+var SYMBOL_SIZE = constants.SYMBOL_SIZE;
+var SYMBOL_STROKE = constants.SYMBOL_STROKE;
+var SYMBOL_SDF = {};
+var SYMBOL_SVG_CIRCLE = Drawing.symbolFuncs[0](SYMBOL_SIZE * 0.05);
+
+function getSymbolSdf(symbol) {
+    if(symbol === 'circle') return null;
+
+    var symbolPath, symbolSdf;
+    var symbolNumber = Drawing.symbolNumber(symbol);
+    var symbolFunc = Drawing.symbolFuncs[symbolNumber % 100];
+    var symbolNoDot = !!Drawing.symbolNoDot[symbolNumber % 100];
+    var symbolNoFill = !!Drawing.symbolNoFill[symbolNumber % 100];
+
+    var isDot = constants.DOT_RE.test(symbol);
```

Comment:
I know this isn't new here, just noticing it: this won't work if `symbol` is provided by number in the first place. Better to just test the number once we have it - ie `isDot = symbolNumber >= 200` (and for [isOpen](https://github.com/plotly/plotly.js/pull/2492/commits/1c6144e291d3d1e191211f4f208672c224bfce4a#diff-202416f5e6d665625197b3b4abf2e142R94) `isOpen = (symbolNumber % 200) >= 100` - perhaps we should make these into helper functions in the `Drawing` module?). Probably not a very well-known feature, but it *is* supported by SVG, and could be useful particularly for symbol arrays, as you could use a typed array.

Dunno if anyone would use both names and numbers in the same plot, but if they did, it would also help to key the `SYMBOL_SDF` cache off the number.

Reply:
Good eye. This probably deserve an issue of its own.